tests/openssl_moretest/enc/tests_ecalls.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <stdlib.h>
+#include "include_openssl.h"
+#include "openssl/crypto.h"
+#include "openssl_moretest_t.h"
+#include "openssl_schema.h"
+
+extern int common_digest_tests(void* buf);
+void ecall_set_rdrand_engine()
+{
+    ENGINE* eng = NULL;
+
+    /* Initialize and opt-in the RDRAND engine. */
+    ENGINE_load_rdrand();
+    eng = ENGINE_by_id("rdrand");
+    if (eng == NULL)
+    {
+        goto done;
+    }
+
+    if (!ENGINE_init(eng))
+    {
+        goto done;
+    }
+
+    if (!ENGINE_set_default(eng, ENGINE_METHOD_RAND))
+    {
+        goto done;
+    }
+
+done:
+
```

Comment:
Does this work given that you always free the `eng` regardless the call succeeds or fails (i.e., if `eng` is freed on return of this function, can the rdrand engine be used afterward)?

Reply:
1. The digest test seems not to require `rdrand eng` because the tests can pass without it. I am thinking about removing the `rdrand eng` for this PR.
2. Some other tests I am preparing have dependency on `rdrand eng`, and it is interesting they can success in this way, that `rdrand eng` being initialized and freed in an ecall called previously. Of course I can split the engine initialization and freeing for the related tests.